crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -176,21 +176,9 @@ impl BitmapInfoHeader {
 
     const NAME: &str = "BITMAPINFOHEADER";
 
-    fn validate_invariants(&self) -> PduResult<()> {
-        check_invariant(self.width.abs() <= u16::MAX.into())
-            .ok_or_else(|| invalid_message_err!("biWidth", "width is too big"))?;
-        check_invariant(self.height.abs() <= u16::MAX.into())
-            .ok_or_else(|| invalid_message_err!("biHeight", "height is too big"))?;
-        check_invariant(self.bit_count <= 32).ok_or_else(|| invalid_message_err!("biBitCount", "invalid bit count"))?;
-
-        Ok(())
-    }
-
     fn encode_with_size(&self, dst: &mut WriteCursor<'_>, size: u32) -> PduResult<()> {
         ensure_fixed_part_size!(in: dst);
 
-        self.validate_invariants()?;
-
```

Comment:
At this point, we don’t have to validate invariants again. They are established when building the type and should be upheld at all time, precisely so that we can rely on them without always checking. (Though it’s okay to perform sanity checks when appropriate. `debug_assert!` are good for that.)